examples/swaggercodegen/api/api/auth.go

```diff
@@ -0,0 +1,17 @@
+package api
+
+import (
+	"net/http"
+	"fmt"
+	"errors"
+)
+
+func AuthenticateRequest(r *http.Request, w http.ResponseWriter) error {
+	apiKey := r.Header.Get("Authorization")
+	if apiKey == "" || apiKey != "apiKeyValue" {
```

Comment:
Is the first part of this necessary?  if it's an empty string, it's not going to be `"apiKeyValue"`